share/eds/accessor_cache.go

```diff
@@ -15,10 +15,71 @@ import (
 )
 
 var (
-	defaultCacheSize = 128
-	errCacheMiss     = errors.New("accessor not found in blockstore cache")
+	errCacheMiss = errors.New("accessor not found in blockstore cache")
 )
 
+// cache is an interface that defines the basic cache operations.
+type cache interface {
+	// Get retrieves an item from the cache.
+	get(shard.Key) (*accessorWithBlockstore, error)
+
+	// getOrLoad attempts to get an item from the cache and, if not found, invokes
+	// the provided loader function to load it into the cache.
+	getOrLoad(ctx context.Context, key shard.Key, loader func(context.Context, shard.Key) (*dagstore.ShardAccessor, error)) (*accessorWithBlockstore, error)
+
+	// enableMetrics enables metrics in cache
+	enableMetrics() error
+}
+
+// multiCache represents a cache that looks into multiple caches one by one.
+type multiCache struct {
```

Comment:
Is the point of this only for metrics? I think its a bit overkill unless we are planning on splitting the cache even further - I would much rather simply have two metrics references saved to the store - its less abstract and cleaner

Reply:
Hmm... on second thought, this is nice because it hides the multiple lookups higher up. But I would still say we should not make it abstract, and just have two caches saved to this multicache instead of an array.